vllm/v1/sample/rejection_sampler.py

```diff
@@ -111,22 +143,82 @@ def forward(
             bonus_token_ids,
             sampling_metadata,
         )
-        return output_token_ids
+
+        return SamplerOutput(
+            sampled_token_ids=output_token_ids,
+            logprobs_tensors=self.get_logprobs_tensors(
+                sampling_metadata,
+                metadata,
+                logits,
+                target_logits if self.return_processed_logprobs else raw_target_logits,
+                bonus_logits,
+                output_token_ids,
+            ),
+        )
+
+    def get_logprobs_tensors(
+        self,
+        sampling_metadata: SamplingMetadata,
+        metadata: SpecDecodeMetadata,
+        logits: torch.Tensor,
+        target_logits: torch.Tensor,
+        bonus_logits: torch.Tensor,
+        sampled_token_ids: torch.Tensor,
+    ) -> LogprobsTensors | None:
+        if not sampling_metadata.max_num_logprobs:
+            return None
+
+        cu_num_sampled_tokens = torch.zeros_like(metadata.cu_num_sampled_tokens)
+        cu_num_sampled_tokens[1:] = metadata.cu_num_sampled_tokens[:-1]
+
+        # Collect target and bonus logits.
+        bonus_logits_indices = metadata.bonus_logits_indices
+        target_logits_indices = metadata.target_logits_indices
+        final_logits = torch.zeros_like(logits, dtype=torch.float32)
+        final_logits[target_logits_indices] = target_logits.to(torch.float32)
+        final_logits[bonus_logits_indices] = bonus_logits.to(torch.float32)
+
+        # Compute accepted token indices.
+        accepted_mask = sampled_token_ids != PLACEHOLDER_TOKEN_ID
+        num_accepted_tokens = accepted_mask.sum(dim=-1)
+        accepted_logit_indices = accepted_mask.nonzero(as_tuple=True)[1]
+        accepted_logit_indices += cu_num_sampled_tokens.repeat_interleave(
+            num_accepted_tokens
+        )
+
+        # Compute logprobs for accepted tokens.
+        return self.compute_logprobs(
+            final_logits[accepted_logit_indices],
+            sampling_metadata.max_num_logprobs,
+            sampled_token_ids[accepted_mask],
+        )
+
+    def compute_logprobs(
```

Comment:
I think this is only called from the method directly above? In which case maybe just inline it?